.github/workflows/inspektor-gadget.yml

```diff
@@ -414,15 +427,19 @@ jobs:
       env:
         KUBERNETES_DISTRIBUTION: "aro"
       with:
+        push: ${{ startsWith(github.ref, 'refs/tags/v') && needs.check-secrets.outputs.container_registry == 'true' }}
         registry: ${{ secrets.CONTAINER_REGISTRY }}
         username: ${{ secrets.CONTAINER_REGISTRY_USERNAME }}
         password: ${{ secrets.CONTAINER_REGISTRY_PASSWORD }}
 
   test-integration:
     name: Integration tests
-    # level: 1
-    needs: [test-unit, build-kubectl-gadget, build-local-gadget, build-docker-image]
+    # level: 2
+    needs: [check-secrets, test-unit, build-kubectl-gadget, build-local-gadget, build-container-image]
     runs-on: ubuntu-latest
+    # It is useless to run the integration tests if the container image was not
+    # pushed to a container registry. They will indeed fail.
+    if: needs.check-secrets.outputs.container_registry == 'true'
```

Comment:
I checked the CI graph because I was afraid that making image building a level 1 job would increase the whole pipeline time and I saw the integration tests were skipped:
https://github.com/kinvolk/inspektor-gadget/actions/runs/2064286450
Was this on purpose to gain development time?

Reply:
It was skipped because we are currently pushing to DockerHub and it does not need to have the `CONTAINER_REGISTRY` secret set. Therefore, I will remove it from [this check](https://github.com/kinvolk/inspektor-gadget/blob/213d485b60d8c4add0a6c63766c907c80c0004aa/.github/workflows/inspektor-gadget.yml#L333) because if the user decides to use DockerHub it will not work.

On the other hand, the whole pipeline time was increased by the time required for its need to be completed, in this case, it is the `check-secrets` job, which means 2 seconds.